File: CS463_MACH1_Demo_CSharp/C463Mach1Samples/ReadParamForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using CSL;
using CSL.Mach1;

namespace CS463Mach1Samples
{
    public partial class ReadParamForm : Form
    {
        public CSLReaderParamControl.MODE mode = CSLReaderParamControl.MODE.NEW;
        public CSLReaderSettings settings;

        public ReadParamForm()
        {
            InitializeComponent();
        }

        private void ReadParamForm_Load(object sender, EventArgs e)
        {
            readerCtrl.SetMode(mode);
            readerCtrl.UpdateForm(settings);
        }

        private void ReadParamForm_FormClosing(object sender, FormClosingEventArgs e)
        {
            settings = readerCtrl.GetSettings();
        }
    }
}

File: CS463_MACH1_Demo_VB/C463Mach1Samples.ConvertedToVBNet/ReadParamForm.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports CSLMach1.CSL
Imports CSLMach1.CSL.Mach1

Public Partial Class ReadParamForm
	Inherits Form
    Public mode As CSLReaderParamControl.MODE = CSLReaderParamControl.MODE.[NEW]
    Public settings As CSLReaderSettings

	Public Sub New()
		InitializeComponent()
	End Sub

	Private Sub ReadParamForm_Load(sender As Object, e As EventArgs)
		readerCtrl.SetMode(mode)
		readerCtrl.UpdateForm(settings)
	End Sub

	Private Sub ReadParamForm_FormClosing(sender As Object, e As FormClosingEventArgs)
		settings = readerCtrl.GetSettings()
	End Sub
End Class
